File: examples/java/java-booking-02_introduce_parameter_object/src/main/java/tv/codely/booking/Customer.java
package tv.codely.booking;

public final class Customer {
    private final CustomerId   id;
    private final CustomerName name;
    private final EmailAddress emailAddress;

    public Customer(CustomerId id, CustomerName name, EmailAddress emailAddress) {
        this.id           = id;
        this.name         = name;
        this.emailAddress = emailAddress;
    }

    public CustomerId id() {
        return id;
    }

    public CustomerName name() {
        return name;
    }

    public EmailAddress emailAddress() {
        return emailAddress;
    }
}

File: examples/csharp/csharp-booking-04_tell_dont_ask/src/Booking/Customer.cs
namespace CodelyTv.Booking
{
    public sealed class Customer
    {
        private readonly CustomerId id;
        private readonly CustomerName name;
        private readonly EmailAddress emailAddress;

        public Customer(CustomerId id, CustomerName name, EmailAddress emailAddress)
        {
            this.id = id;
            this.name = name;
            this.emailAddress = emailAddress;
        }
    }
}
